### ISSUE TITLE
Remove merge_highlights_into_result

### ISSUE THREAD

**Issue by albertisfu** [freelawproject/courtlistener#3670]
Currently, we use `merge_highlights_into_result` to merge highlights on fields that could match terms from both the normal and the exact versions of the field.

This method is no longer used in RECAP or Opinions Search, as they now employ `match_fields` to return the combined highlights.

### To completely remove this method, we would need to:

Create a PR for:

- Adding support for `term_vectors` on fields that require highlighting in the Parenthetical, Oral Arguments, and Person mapping.
- Updating tests for the new FVH approach.
- Continuing to use plain highlighting in production for these fields.
This should be straightforward and would take just a couple of hours.

- Deploy the changes. 
  Within a pod, rename the `parenthetical_group`, `oral_arguments`, `oral_arguments_percolator` and `people_db_index` to their vector version. 
  e.g:
  `parenthetical_group_vectors`, `oral_arguments_vectors`, `oral_arguments_percolator_vectors` and `people_db_index_vectors`

- Create a new index for each of the following:

```
manage.py search_index --create --models search.ParentheticalGroup
manage.py search_index --create --models people_db.Person
manage.py search_index --create --models audio.Audio
manage.py search_index --create --models  alerts.Alert
```

- Perform a `re_index` from each index to their new vector version:
```
POST
/_reindex
{
    "max_docs":10000000, #Number of documents we want to copy.
    "source":{
        "index":"oral_arguments"
    },
    "dest":{
        "index":"oral_arguments_vectors"
    }
}
```

- Once this is ready.
  I can submit another PR that uses the new indices, enables FVH for these documents, and removes `merge_highlights_into_result`.

**Comment by mlissner** [comment]
Great. Let's get this done while we're thinking about it. Thanks.
